sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -439,6 +440,16 @@ class Analyzer(
             case s: Star => s.expand(child, resolver)
             case o => o :: Nil
           })
+        case p: Murmur3Hash if containsStar(p.children) =>
```

Comment:
ah yea, then we have 6 cases now, we should find a way to replace the `children` for them, maybe we can add a `TreeNode.replaceAllChildren` to do it.